product_supplierinfo_import/wizards/product_supplierinfo_import.py

```diff
@@ -16,17 +17,42 @@ class ProductSupplierInfoImport(models.TransientModel):
     _name = "product.supplierinfo.import"
     _description = "Import supplier info records"
 
-    supplier_id = fields.Many2one(comodel_name="res.partner", required=True)
+    supplier_id = fields.Many2one(
+        "res.partner",
+        string="Supplier",
+        related="template_id.supplier_id",
```

Comment:
This can't be related, as it will change the previous logic. Better use a compute storable field that depends on the template